src/shogun/util/converters.h

```diff
@@ -56,6 +56,9 @@ namespace shogun
 		    std::is_signed<I>::value && std::is_unsigned<J>::value, I>
 		safe_convert(J value)
 		{
+			if constexpr (std::is_same_v<bool, J>)
```

Comment:
@lisitsyn better idea? and yes `bool` is apparently unsigned ;)

Reply:
ok this doesn't really annulled the compiler warning... i guess i need to do a type trait for the input type :S